rust_dev_preview/ecr/src/bin/list-images.rs

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_config::meta::region::RegionProviderChain;
+use aws_sdk_ecr::{Error, Region, PKG_VERSION};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The AWS Region.
+    #[structopt(short, long)]
+    region: Option<String>,
+
+    /// The name of the repository.
+    #[structopt(long)]
+    repository: String,
+
+    /// Whether to display additional information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+// Show the images in a repository.
+// snippet-start:[ecr.rust.list-images]
+async fn show_images(
+    client: &aws_sdk_ecr::Client,
+    repository: &str,
+) -> Result<(), aws_sdk_ecr::Error> {
+    let rsp = client
+        .list_images()
+        .repository_name(repository)
+        .send()
+        .await?;
+
+    let images = rsp.image_ids.unwrap_or_default();
+
+    println!("found {} images", images.len());
+
+    for image in images {
+        println!(
+            "image: {}:{}",
+            image.image_tag.unwrap(),
```

Comment:
probably needs an accessor? Not sure on this one. Should try it out and see if it works.

Reply:
It works fine.